walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,58 @@
 
 package com.google.sps;
 
+import java.util.Comparator;
 import java.util.Collection;
+import static java.util.Collections.disjoint;
+import static java.util.Collections.sort;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      Collection<TimeRange> queryResult = new ArrayList<>();
+      List<Event> eventsList = new ArrayList<>(events);
+      sort(eventsList, new Comparator<Event>() {
+        public int compare (Event e1, Event e2) {
+            return TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen());
+          }
+        });
+      eventsList.removeIf(e -> (
+        e.getWhen().duration() <= 0 || disjoint(e.getAttendees(), request.getAttendees())));
+      if( request.getAttendees().isEmpty()){
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+      if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+          return queryResult;
```

Comment:
You probably just want to return Collections.EMPTY_LIST.  If you return queryResult the reader has to scan above to see if anything got added.

Reply:
Done.